test/functional/discovered_hosts_controller_test.rb

```diff
@@ -95,19 +95,16 @@ def test_edit_form_quick_submit
   end
 
   def test_edit_form_submit_parameters
-    disable_taxonomies do
-      host = Host::Discovered.import_host(@facts)
-      domain = FactoryGirl.create(:domain)
-      hostgroup = FactoryGirl.create(:hostgroup, :with_subnet, :with_environment, :with_rootpass, :with_os, :domain => domain)
-      get :edit, {
-        :id => host.id,
-        :host => {
-          :hostgroup_id => hostgroup.id
-        } }, set_session_user_default_manager
-
-      assert_select '#host_operatingsystem_id [selected]' do |elements|
-        assert_equal hostgroup.operatingsystem.id.to_s, elements.first[:value]
-      end
+    host = Host::Discovered.import_host(@facts)
+    domain = FactoryGirl.create(:domain)
+    hostgroup = FactoryGirl.create(:hostgroup, :with_subnet, :with_environment, :with_rootpass, :with_os, :domain => domain)
+    get :edit, {
+      :id => host.id,
+      :host => {
+        :hostgroup_id => hostgroup.id
+      } }, set_session_user_default_manager
+    assert_select '#host_operatingsystem_id [selected]' do |elements|
+      assert_equal hostgroup.operatingsystem.id.to_s, elements.first[:value]
     end
```

Comment:
Can you also add asserts for:

* puppet environment, ca, master
* architecture, media, ptable

This covers all problematic ones. I think we might have more issues than OS - I just saw we also need view_ permissions for all the others.